server/src/main/java/com/thoughtworks/go/config/PartialConfigParseResult.java

```diff
@@ -17,52 +17,97 @@
 package com.thoughtworks.go.config;
 
 import com.thoughtworks.go.config.remote.PartialConfig;
+import com.thoughtworks.go.domain.materials.Modification;
 
 import java.util.Objects;
 
+/*
+    Understands the parse results of a config repo materials
+
+    @field latestParsedModification Modification
+    Latest parsed modification represents the latest known commit from the config repo material
+
+    @field goodModification Modification
+    Good modification represents the last commit from which the partials configurations were fetched successfully.
+
+    @field partialConfig PartialConfig
+    partial config represents pipelines and environments fetched from the good modification
+
+    @field exception Exception
+    Any expect occurred while parsing config repository.
+*/
+
+
 public class PartialConfigParseResult {
-    private final String revision;
-    private PartialConfig lastSuccess;
-    private Exception lastFailure;
+    private Modification latestParsedModification;
+    private Modification goodModification;
+    private PartialConfig partialConfig;
+    private Exception exception;
 
-    public PartialConfigParseResult(String revision, PartialConfig newPart) {
-        this.revision = revision;
-        this.lastSuccess = newPart;
+    public PartialConfigParseResult(Modification latestParsedModification, Modification goodModification, PartialConfig partialConfig, Exception exception) {
+        this.latestParsedModification = latestParsedModification;
+        this.goodModification = goodModification;
+        this.partialConfig = partialConfig;
+        this.exception = exception;
     }
 
-    public PartialConfigParseResult(String revision, Exception ex) {
-        this.revision = revision;
-        this.lastFailure = ex;
+    public Modification getLatestParsedModification() {
+        return latestParsedModification;
     }
 
-    public boolean isSuccessful() {
-        return null == getLastFailure();
+    public Modification getGoodModification() {
+        return goodModification;
     }
 
-    public PartialConfig getLastSuccess() {
-        return lastSuccess;
+    public PartialConfig getPartialConfig() {
```

Comment:
Not a big deal because one can figure it out by the context in which this is used, but perhaps rename the field to `lastGoodPartialConfig()` might be better. When I was looking over the `ConfigReposMaterialsParseResultManager.parseFailed()` method, I had wondered why we persisted the config from `existingResult -> newResult`, but later realized that it was the last known good result. Minor nitpick, but slightly more descriptive naming can be helpful.

Reply:
Hello @marques-work, I agree with you on the descriptive naming for methods.
As of now, I've changed `getPartialConfig` -> `lastGoodPartialConfig`.

For the names, `ConfigReposMaterialsParseResultManager.parseFailed()` and `ConfigReposMaterialsParseResultManager.parseSuccess()`. The reason for just calling it `parseFailed` or `parseSuccess` is that, it could be failed for the first time or successive times, and we dont know that information.

As of now, I am merging this. please let me know if  any of the names can be improved.